### ISSUE TITLE
Groups

### ISSUE THREAD

**Issue by rodrigosiviero** [sergiocasero/sweet_home_3d_ha_lightning#8]
It's a good idea of a plugin, but would be better if it can export per group of lighting, for instance, I have a front yard light composed of 6 lights, I don't need 64 images, just two on and off, thinking about gangs/scenes instead of individual lights.
Maybe use a prefix for groups so it can scan manually like - group.light.frontyard

**Comment by guimatheus92** [comment]
> It's a good idea of a plugin, but would be better if it can export per group of lighting, for instance, I have a front yard light composed of 6 lights, I don't need 64 images, just two on and off, thinking about gangs/scenes instead of individual lights. Maybe use a prefix for groups so it can scan manually like - group.light.frontyard

Hi @rodrigosiviero! Eu vi que é brasileiro, eu estou criando um projeto parecido com este que também vai exportar as imagems automaticamente, com mais features. Já que este projeto foi abandonado pelo criador. Você poderia elaborar mais a sua ideia por favor?

**Comment by rodrigosiviero** [comment]
> > It's a good idea of a plugin, but would be better if it can export per group of lighting, for instance, I have a front yard light composed of 6 lights, I don't need 64 images, just two on and off, thinking about gangs/scenes instead of individual lights. Maybe use a prefix for groups so it can scan manually like - group.light.frontyard
> 
> Hi @rodrigosiviero! Eu vi que é brasileiro, eu estou criando um projeto parecido com este que também vai exportar as imagems automaticamente, com mais features. Já que este projeto foi abandonado pelo criador. Você poderia elaborar mais a sua ideia por favor?

Basicamente oque disse é, eu nao preciso de 1 imagem por lampada acesa, isso e maluquice, voce pode ter 1 lampada smart ou voce pode ter 1 interruptor smart que acendem 10 lampadas, entao o correto seria agrupar as lampadas que acendem junto e criar a imagem com isso, se a lampada é individual beleza mas se ela ta num grupo precisa acionar o grupo todo

**Comment by guimatheus92** [comment]
> > > It's a good idea of a plugin, but would be better if it can export per group of lighting, for instance, I have a front yard light composed of 6 lights, I don't need 64 images, just two on and off, thinking about gangs/scenes instead of individual lights. Maybe use a prefix for groups so it can scan manually like - group.light.frontyard
> > 
> > 
> > Hi @rodrigosiviero! Eu vi que é brasileiro, eu estou criando um projeto parecido com este que também vai exportar as imagems automaticamente, com mais features. Já que este projeto foi abandonado pelo criador. Você poderia elaborar mais a sua ideia por favor?
> 
> Basicamente oque disse é, eu nao preciso de 1 imagem por lampada acesa, isso e maluquice, voce pode ter 1 lampada smart ou voce pode ter 1 interruptor smart que acendem 10 lampadas, entao o correto seria agrupar as lampadas que acendem junto e criar a imagem com isso, se a lampada é individual beleza mas se ela ta num grupo precisa acionar o grupo todo

Sim, faz sentido. Estou com uma versão estável aqui gerando as imagens certinho, mas no meu caso o meu projeto tem 62 luzes e isso geraria 4,611,686,018,427,387,904 de imagens. É insustentável realmente. Vou lançar essa versão no ar e tentar adicionar uma feature pro usuario "selecionar/criar" grupos de luzes.

**Comment by rodrigosiviero** [comment]
> > > > It's a good idea of a plugin, but would be better if it can export per group of lighting, for instance, I have a front yard light composed of 6 lights, I don't need 64 images, just two on and off, thinking about gangs/scenes instead of individual lights. Maybe use a prefix for groups so it can scan manually like - group.light.frontyard
> > > 
> > > 
> > > Hi @rodrigosiviero! Eu vi que é brasileiro, eu estou criando um projeto parecido com este que também vai exportar as imagems automaticamente, com mais features. Já que este projeto foi abandonado pelo criador. Você poderia elaborar mais a sua ideia por favor?
> > 
> > 
> > Basicamente oque disse é, eu nao preciso de 1 imagem por lampada acesa, isso e maluquice, voce pode ter 1 lampada smart ou voce pode ter 1 interruptor smart que acendem 10 lampadas, entao o correto seria agrupar as lampadas que acendem junto e criar a imagem com isso, se a lampada é individual beleza mas se ela ta num grupo precisa acionar o grupo todo
> 
> Sim, faz sentido. Estou com uma versão estável aqui gerando as imagens certinho, mas no meu caso o meu projeto tem 62 luzes e isso geraria 4,611,686,018,427,387,904 de imagens. É insustentável realmente. Vou lançar essa versão no ar e tentar adicionar uma feature pro usuario "selecionar/criar" grupos de luzes.

exato, minha casa nova tem 140 luzes e mais de 40 switches inteligentes ia ser impossivel usar isso aqui sem agrupamento

**Comment by guimatheus92** [comment]
> > > > > It's a good idea of a plugin, but would be better if it can export per group of lighting, for instance, I have a front yard light composed of 6 lights, I don't need 64 images, just two on and off, thinking about gangs/scenes instead of individual lights. Maybe use a prefix for groups so it can scan manually like - group.light.frontyard
> > > > 
> > > > 
> > > > Hi @rodrigosiviero! Eu vi que é brasileiro, eu estou criando um projeto parecido com este que também vai exportar as imagems automaticamente, com mais features. Já que este projeto foi abandonado pelo criador. Você poderia elaborar mais a sua ideia por favor?
> > > 
> > > 
> > > Basicamente oque disse é, eu nao preciso de 1 imagem por lampada acesa, isso e maluquice, voce pode ter 1 lampada smart ou voce pode ter 1 interruptor smart que acendem 10 lampadas, entao o correto seria agrupar as lampadas que acendem junto e criar a imagem com isso, se a lampada é individual beleza mas se ela ta num grupo precisa acionar o grupo todo
> > 
> > 
> > Sim, faz sentido. Estou com uma versão estável aqui gerando as imagens certinho, mas no meu caso o meu projeto tem 62 luzes e isso geraria 4,611,686,018,427,387,904 de imagens. É insustentável realmente. Vou lançar essa versão no ar e tentar adicionar uma feature pro usuario "selecionar/criar" grupos de luzes.
> 
> exato, minha casa nova tem 140 luzes e mais de 40 switches inteligentes ia ser impossivel usar isso aqui sem agrupamento

Seu projeto deve estar top então hein, parabéns! Eu to com uma versão aqui com agrupamento, queria que você testasse. Porque mesmo criando grupos, ainda teria imagem pra cacete (muito menos que a versão sem grupos obviamente), no meu caso ficaria assim:

![image](https://github.com/sergiocasero/sweet_home_3d_ha_lightning/assets/11640754/a33eccef-5759-4766-a7a6-21f6a2a6efeb)

Mas será que ficaria bom na imagem ver todas acesas? Não sei se tem um jeito melhor pra fazer também. Se puder, você poderia fazer um teste por favor? Acho que você tem um cenário ideal, assim como eu.

Se tiver alguma dúvida, abre uma issue lá no meu repo que a gente vai conversando.

A versão com grupos é essa aqui: `https://github.com/guimatheus92/SweetHome3D_HomeAssistant_Lightning/releases/tag/v2.0.0-beta`
